src/ansiblelint/rules/var_naming.py

```diff
@@ -33,6 +33,53 @@ class VariableNamingRule(AnsibleLintRule):
     re_pattern_str = options.var_naming_pattern or "^[a-z_][a-z0-9_]*$"
     re_pattern = re.compile(re_pattern_str)
     reserved_names = get_reserved_names()
+    # List of special variables that should be treated as read-only. This list
+    # does not include connection variables, which we expect users to tune in
+    # specific cases.
+    # https://docs.ansible.com/ansible/latest/reference_appendices/special_variables.html
+    read_only_names = {
+        "ansible_check_mode",
+        "ansible_collection_name",
+        "ansible_config_file",
+        "ansible_dependent_role_names",
+        "ansible_diff_mode",
+        "ansible_facts",  # special fact
```

Comment:
this is actually updatable by user on purpose .. so they can 'fake facts' if needed

Reply:
Thanks, I moved it to the allowed list.